### PR TITLE
Add Role to VehicleClassification

### PR DESCRIPTION
**Pull request by adrianschultz**
﻿#### Reference to a related issue in the repository
[Issue](https://github.com/OpenSimulationInterface/open-simulation-interface/issues/584)

#### Add a description
Added Role to VehicleClassification in alignment with OSC 1.2

#### Take this checklist as orientation for yourself, if this PR is ready for the Change Control Board:
- [x] My suggestion follows the [style and contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html).
- [x] I have taken care about the [documentation](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/commenting.html).
- [x] I have done the [DCO signoff](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html#developer-certification-of-origin-dco).
- [x] My changes generate no errors when passing CI tests. 
- [x] I have successfully implemented and tested my fix/feature locally.
- [x] Appropriate reviewer(s) are assigned.

If you can’t check all of them, please explain why.
If all boxes are checked or commented and you have achieved at least one positive review, you can assign the label ReadyForCCBReview!

### CODE REVIEW

**Comment by adrianschultz** [comment]
The failing builds are caused by a missing blank line at the end of the changed file. I will change that after this PR was discussed in the Harmonisation group.

File: osi_object.proto [hunk 1]
```diff
@@ -950,5 +954,46 @@ message MovingObject
...
+
+            // Other (unspecified but known) role of vehicle.
+            //
+            ROLE_OTHER = 1;
+
+            // The vehicle role is ambulance.
+            //
+            ROLE_AMBULANCE = 2;
+
+            // The vehicle role is civil.
```
Comment (kmeids) [inline osi_object.proto:974]: **Output CCB 16.02.22:**
1. Please provide better documentation for "ROLE_CIVIL"

**Comment by kmeids** [comment]
**Output CCB 16.02.22:**
1. Please take a look on the provided comments.
2. Which value is supposed to represent "normal" traffic participants? This should not be ROLE_OTHER  or ROLE_UNKNOWN.

**Comment by HendrikAmelunxen** [comment]
Output of harmonization meeting (08.04.22): The roles need more clarification, e.g. what role gets a fire truck without emergency lights? Civil or fire? Same question for police, road assistance, and ambulance. 
Is an additional role for garbage trucks needed, since they have a specific driving behaviour when they are in action?

**Comment by adrianschultz** [comment]
> Output of harmonization meeting (08.04.22): The roles need more clarification, e.g. what role gets a fire truck without emergency lights? Civil or fire? Same question for police, road assistance, and ambulance. Is an additional role for garbage trucks needed, since they have a specific driving behaviour when they are in action?

Wether those are active or not can be looked up in VehicleClassification / LightState. So, I would suggest to set the role in your example to ROLE_FIRE. I can add a clarification in the enum description to explain this.

Adding new roles for garbage related vehicles (truck, sweeper etc.) and road construction related vehicles would be reasonable in my opinion.
e.g. :
// The vehicle role is garbage collection, e.g. a garbage truck or sweeper.
//
ROLE_GARBAGE_COLLECTION = 9;

// The vehicle role is road construction, e.g. a excavator or tipper truck.
//
ROLE_ROAD_CONSTRUCTION = 10;

File: osi_object.proto [hunk 2]
```diff
@@ -950,5 +954,64 @@ message MovingObject
...
+            //
+            ROLE_UNKNOWN = 0;
+
+            // Other (unspecified but known) role of vehicle.
+            //
+            ROLE_OTHER = 1;
+
+            // The vehicle role is ambulance.
+            //
+            ROLE_AMBULANCE = 2;
```
Comment (stefancyliax) [inline osi_object.proto:982]: OSI CCB 23.05. 
- please change order of Ambulance and Civil. The standard case should be the first one.

**Comment by stefancyliax** [comment]
OSI CCB 23.05.2022:
Topic will be finally discussed on 30.05..
Decision about merging then. If it is ready for merging, it will be included in OSI 3.5

File: osi_object.proto (hunk 2 above)
Comment (adrianschultz) [reply osi_object.proto]: Change is applied.

**Comment by stefancyliax** [comment]
OSI CCB:
- There happend something with rebasing, that has to be cleaned up. @pmai to look at this.
- @HendrikAmelunxen What was the result of the meeting on 25.05.2022? Is this ready to be merged?

File: osi_object.proto [hunk 3]
```diff
@@ -1005,5 +1009,64 @@ message MovingObject
...
+        // Definition of vehicle roles. 
+        //
+        // The role specifies the perceived role of a vehicle. Therefore, a 
+        // civil police car is set to ROLE_CIVIL as long as it cannot be 
+        // perceived as a ROLE_POLICE. 
+        // The normal use case will consist of 99% ROLE_CIVIL, as this role is 
+        // used for normal vehicles.
+        // To check whether a role is "active" (e.g. blue flashing light of an 
+        // ambulance or an amber flashing light of a sweeper) can be looked up 
+        // in message LightState.
```
Comment (stefancyliax) [inline osi_object.proto:1022]: OSI CCB 30.05.2022:
```suggestion
        // The role specifies the perceived role of a vehicle. Therefore, a 
        // civil police car is set to ROLE_CIVIL as long as it cannot be 
        // perceived as a ROLE_POLICE. 
        // The normal use case will consist of 99% ROLE_CIVIL, as this role is 
        // used for normal vehicles.
        // These roles may influence the behavior in traffic of the surrounding vehicles.
        // To check whether a role is "active" (e.g. blue flashing light of an 
        // ambulance or an amber flashing light of a sweeper) can be looked up 
        // in message LightState.
```

**Review by pmai — APPROVED** [review]

**Pull request merged**